File: ui/src/api/messages.ts
import { authAxios } from "@/lib/axios-instance";

export const getMessageByID = async (messageID: string) => {
  const response = await authAxios.post(`/messages/solo/${messageID}`);
  return response.data;
};

export const getMessagesByProjectID = async (projectID: string) => {
  const response = await authAxios.get(`/messages/${projectID}`);
  return response.data;
};

File: api/routes/messages.go
package routes

import (
	"ai-zustack/api/handlers"
	"ai-zustack/api/middleware"

	"github.com/gofiber/fiber/v2"
)

func MessagesRoutes(app *fiber.App) {
	app.Post("/messages/solo/:messageID", middleware.User, handlers.GetMessageByID)
	app.Post("/webhook/messages/:projectID/:model", middleware.User, handlers.WebhookMessage)
	app.Get("/messages/:projectID", middleware.User, handlers.GetMessagesByProjectID)
}
